libbpf-rs/src/object.rs

```diff
@@ -139,3 +225,46 @@ pub enum ProgramType {}
 /// Attach type of a [`Program`]. Maps to `enum bpf_attach_type` in kernel uapi.
 #[non_exhaustive]
 pub enum ProgramAttachType {}
+
+/// Represents a loaded [`Program`].
+///
+/// The kernel ensure the atomicity and safety of operations on a `LoadedProgram`. Therefore,
+/// this handle is safe to clone and pass around between threads. This is essentially a
+/// file descriptor.
+///
+/// If you attempt to attach a `LoadedProgram` with the wrong attach method, the `attach_*`
+/// method will fail with the appropriate error.
+#[derive(Clone)]
+pub struct LoadedProgram {}
+
+impl LoadedProgram {
+    pub fn name(&self) -> &str {
+        unimplemented!();
+    }
+
+    /// Name of the section this `Program` belongs to.
+    pub fn section(&self) -> &str {
+        unimplemented!();
+    }
+
+    pub fn prog_type(&self) -> ProgramType {
+        unimplemented!();
+    }
+
+    /// Returns a file descriptor to the underlying program.
+    pub fn fd(&self) -> i64 {
+        unimplemented!();
+    }
+
+    pub fn attach_type(&self) -> ProgramAttachType {
+        unimplemented!();
+    }
+
+    pub fn attach_cgroup<T: AsRef<str>>(&mut self, _cgroup: T) -> Result<Link> {
+        unimplemented!();
+    }
+
+    pub fn attach_perf_event(&mut self, _pfd: i64) -> Result<Link> {
```

Comment:
fd is not going to be bigger than 32-bits, no need for i64 here

Reply:
ok